public/app/core/components/AppChrome/TopBar/SingleTopBar.tsx

```diff
@@ -18,100 +21,113 @@ import { buildBreadcrumbs } from '../../Breadcrumbs/utils';
-            onClick={onToggleMegaMenu}
-            tooltip={t('navigation.megamenu.open', 'Open menu')}
-          >
-            <Stack gap={0} alignItems="center">
-              <Branding.MenuLogo className={styles.img} />
-              <Icon size="sm" name="angle-down" />
-            </Stack>
-          </ToolbarButton>
-        )}
-        <Breadcrumbs breadcrumbs={breadcrumbs} className={styles.breadcrumbsWrapper} />
-      </Stack>
+    <>
+      <div className={styles.layout}>
+        <Stack minWidth={0} gap={0.5} alignItems="center" flex={{ xs: 2, lg: 1 }}>
+          {!menuDockedAndOpen && (
+            <ToolbarButton
+              narrow
+              id={MEGA_MENU_TOGGLE_ID}
+              onClick={onToggleMegaMenu}
+              tooltip={t('navigation.megamenu.open', 'Open menu')}
+            >
+              <Stack gap={0} alignItems="center">
+                <Branding.MenuLogo className={styles.img} />
+                <Icon size="sm" name="angle-down" />
+              </Stack>
+            </ToolbarButton>
+          )}
+          <Breadcrumbs breadcrumbs={breadcrumbs} className={styles.breadcrumbsWrapper} />
+          {!showToolbarLevel && breadcrumbActions}
+        </Stack>
 
-      <Stack gap={0.5} alignItems="center">
-        <TopSearchBarCommandPaletteTrigger />
-        {unifiedHistoryEnabled && <HistoryContainer />}
-        <QuickAdd />
-        {enrichedHelpNode && (
-          <Dropdown overlay={() => <TopNavBarMenu node={enrichedHelpNode} />} placement="bottom-end">
-            <ToolbarButton iconOnly icon="question-circle" aria-label={t('navigation.help.aria-label', 'Help')} />
-          </Dropdown>
-        )}
-        <ToolbarButton
-          icon="monitor"
-          className={styles.kioskToggle}
-          onClick={onToggleKioskMode}
-          tooltip={t('navigation.kiosk.tooltip', 'Enable kiosk mode')}
-        />
-        {!contextSrv.user.isSignedIn && <SignInLink />}
-        {config.featureToggles.inviteUserExperimental && <InviteUserButton />}
-        {config.featureToggles.extensionSidebar && <ExtensionToolbarItem />}
-        {profileNode && <ProfileButton profileNode={profileNode} />}
-      </Stack>
-    </div>
+        <Stack
+          gap={0.5}
+          alignItems="center"
+          justifyContent={'flex-end'}
+          flex={1}
+          data-testid={!showToolbarLevel ? Components.NavToolbar.container : undefined}
+          minWidth={{ xs: 'unset', lg: 0 }}
+        >
+          <TopSearchBarCommandPaletteTrigger />
+          {unifiedHistoryEnabled && !isSmallScreen && <HistoryContainer />}
+          {!isSmallScreen && <QuickAdd />}
+          {enrichedHelpNode && (
+            <Dropdown overlay={() => <TopNavBarMenu node={enrichedHelpNode} />} placement="bottom-end">
+              <ToolbarButton iconOnly icon="question-circle" aria-label={t('navigation.help.aria-label', 'Help')} />
+            </Dropdown>
+          )}
+          <NavToolbarSeparator />
```

Comment:
I looked briefly and I can't see why, but I get an extra separator on non-dashboards pages (such as /admin)

![image](https://github.com/user-attachments/assets/83696e70-e3ef-4c2b-8de7-0d33a3e3e86a)

I have toggle disabled.

Reply:
@joshhunt good catch! The invite button had double separator lines inside it. Fixed